docs/diagrams/ui-sequence-diagram.puml

```diff
@@ -0,0 +1,40 @@
+@startuml
+activate ":Main" #FFBBBB
+":Main" -> ":UI": readUserInput()
+activate ":UI" #FFBBBB
+
+
+":UI" -> User: Gets user input
+activate "User" #FFBBBB
+User --> ":UI": "userInput: String"
+deactivate
+":UI" --> ":Main": "userInput: String"
+deactivate
+
+create ":Parser"
+":Main" -> ":Parser": new Parser()
+activate ":Parser" #FFBBBB
+":Parser" --> ":Main": "p: Parser"
+
+":Main" -> ":Parser": p.parse(userInput: String)
```

Comment:
You might want to break down your complicated diagram here into subdiagrams. This can be done using ref frames
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/93811746/09c7ec17-ccd3-462d-9f4d-865599343626)